contracts/modules/uniswap/v3/BytesLib.sol

```diff
@@ -44,4 +44,12 @@ library BytesLib {
             res.length := calldataload(lengthPtr)
         }
     }
+
+    function toBytesArray(bytes calldata _bytes, uint256 arg) internal pure returns (bytes[] calldata res) {
```

Comment:
Maybe dumb q but whats the difference between this and the `toBytes` func above? Just the return type?

Reply:
Not a stupid question at all - its just the return type indeed. `toBytes`, `toAddressArray` and `toBytesArray` are all the same just different return types lol

Reply:
possible to have toBytesArray and toAddress just reuse toBytes and force cast? Think worst case you can force cast with assembly walrus haha.. maybe adds a bit of gas for a pointer dup tho